src/devices/src/virtio/net/device.rs

```diff
@@ -761,49 +834,103 @@ mod tests {
             }
 
-            let queues = QUEUE_SIZES.iter().map(|&s| Queue::new(s)).collect();
+            let queues = Vec::new();
 
-            let mmds_ns = if allow_mmds_requests {
-                Some(MmdsNetworkStack::new_with_defaults())
-            } else {
-                None
-            };
+            let mmds_ns = Some(MmdsNetworkStack::new_with_defaults());
 
-            Ok(Net {
+            Net {
                 tap,
                 avail_features,
                 acked_features: 0u64,
-                mem,
+                mem: Net::default_guest_memory(),
                 queues,
                 queue_evts,
-                rx_rate_limiter,
-                tx_rate_limiter,
+                rx_rate_limiter: RateLimiter::default(),
+                tx_rate_limiter: RateLimiter::default(),
                 rx_deferred_frame: false,
                 rx_deferred_irqs: false,
                 rx_bytes_read: 0,
                 rx_frame_buf: [0u8; MAX_BUFFER_SIZE],
                 tx_frame_buf: [0u8; MAX_BUFFER_SIZE],
                 tx_iovec: Vec::with_capacity(QUEUE_SIZE as usize),
                 interrupt_status: Arc::new(AtomicUsize::new(0)),
-                interrupt_evt: EventFd::new(libc::EFD_NONBLOCK).map_err(Error::EventFd)?,
+                interrupt_evt: EventFd::new(libc::EFD_NONBLOCK).unwrap(),
                 guest_mac: Some(MacAddr::from_bytes_unchecked(
                     &config_space.clone()[..MAC_ADDR_LEN],
                 )),
                 device_activated: false,
                 config_space,
                 mmds_ns,
-            })
-
+                test_mutators,
+            }
         }
 
         fn default_guest_mac() -> MacAddr {
-            unimplemented!()
+            MacAddr::parse_str("11:22:33:44:55:66").unwrap()
+        }
+
+        fn default_guest_memory() -> GuestMemory {
+            GuestMemory::new(&[(GuestAddress(0), 0x10000)]).unwrap()
+        }
+
+        fn rx_single_frame_no_irq_coalescing(&mut self) -> bool {
+            let ret = self.rx_single_frame();
+            if self.rx_deferred_irqs {
+                self.rx_deferred_irqs = false;
+                let _ = self.signal_used_queue();
+            }
+            ret
+        }
+
+        // Returns handles to virtio queues creation/activation and manipulation.
+        fn virtqueues(mem: &GuestMemory) -> (VirtQueue, VirtQueue) {
+            let rxq = VirtQueue::new(GuestAddress(0), mem, 16);
+            let txq = VirtQueue::new(GuestAddress(0x1000), mem, 16);
+            assert!(rxq.end().0 < txq.start().0);
+
+            (rxq, txq)
+        }
+
+        fn set_mac(&mut self, mac: MacAddr) {
+            self.guest_mac = Some(mac);
+            let mut config_space;
+            config_space = Vec::with_capacity(MAC_ADDR_LEN);
+            // This is safe, because we know the capacity is large enough.
+            unsafe { config_space.set_len(MAC_ADDR_LEN) }
```

Comment:
You can use the `vec!` macro instead which guarantees that it allocates exactly `MAC_ADDR_LEN` bytes. In this case capacity will be equal to len and you won't need to use the unsafe `set_len` function.

Reply:
I think these lines also show somewhere else in the code. Can you also replace those pls?